client.go

```diff
@@ -393,6 +393,9 @@ type Backoff func(min, max time.Duration, attemptNum int, resp *http.Response) t
 // attempted. If overriding this, be sure to close the body if needed.
 type ErrorHandler func(resp *http.Response, err error, numTries int) (*http.Response, error)
 
+// PrepareRetry is called before retry operation. It can be used to for example re-sign the request
```

Comment:
``
```suggestion
// PrepareRetry is called before retry operation. It can be used for example to re-sign the request
```
or ```For example, it can be used to re-sign the request```
Minor thing but the order is odd